log4j-core/src/main/java/org/apache/logging/log4j/core/util/TypeUtil.java

```diff
@@ -53,9 +54,7 @@ private TypeUtil() {
     public static List<Field> getAllDeclaredFields(Class<?> cls) {
         final List<Field> fields = new ArrayList<>();
         while (cls != null) {
-            for (final Field field : cls.getDeclaredFields()) {
-                fields.add(field);
-            }
+            fields.addAll(Arrays.asList(cls.getDeclaredFields()));
```

Comment:
I have the impression that we are adding an extra GC load to a commonly used part of the code base. I'd prefer the earlier approach. Would you mind reverting this, please?

Reply:
Done.
TY

Reply:
I don't think it's entirely clear which one will perform better, but I agree with @vy that the original is likely to outperform addAll.

The benefit of addAll is that the receiving collection doesn't need to resize the underlying array multiple times, however ArrayList.addAll(Collection) calls Collection.toArray(), and Arrays.asList(arr) uses an implementation which produces a copy of the underlying array on each call. That's expensive, but we're deep enough into hot methods that C2 may be able to help us out by avoiding the duplicate array allocation (potentially including the getDeclaredFields() array).